unittesting/utils/reloader.py

```diff
@@ -57,6 +102,27 @@ def reload_package(pkg_name, dummy=True, verbose=True):
         dprint("end", fill='-')
 
 
+def reload_dependency(dependency_name, dummy=True, verbose=True):
+    """
+    Package Control dependencies aren't regular packages, so we don't want to
+    call `sublime_plugin.unload_module` or `sublime_plugin.reload_plugin`.
+    Instead, we manually unload all of the modules in the dependency and then
+    `reload_package` any packages that use that dependency. (We have to manually
+    unload the dependency's modules because calling `reload_package` on a
+    dependent module will not unload the dependency.)
+    """
+    for name in get_package_modules(dependency_name):
+        del sys.modules[name]
+
+    manager = PackageManager()
+    for package in manager.list_packages():
+        if dependency_name in manager.get_dependencies(package):
```

Comment:
I think we should cache the list coming out of `manager.get_dependencies(package)` converting it to a set. For example:
```python
    manager = PackageManager()
    dependencies = set(manager.get_dependencies(package))
    for package in manager.list_packages():
        if dependency_name in dependencies:
```